firmware/src/moving_light.c

```diff
@@ -0,0 +1,359 @@
+    case PID_LAMP_STATE:
+      return RDMResponder_GenericGetUInt8(header, g_moving_light.lamp_state);
+    case PID_LAMP_ON_MODE:
+      return RDMResponder_GenericGetUInt8(header, g_moving_light.lamp_on_mode);
+    case PID_DISPLAY_INVERT:
+      return RDMResponder_GenericGetUInt8(
+          header, g_moving_light.display_invert);
+    case PID_DISPLAY_LEVEL:
+      return RDMResponder_GenericGetUInt8(header, g_moving_light.display_level);
+    case PID_POWER_STATE:
+      return RDMResponder_GenericGetUInt8(header, g_moving_light.power_state);
+    default:
+      return RDM_RESPONDER_NO_RESPONSE;
+  }
+}
+
+int MovingLightModel_SetUint8(const RDMHeader *header,
+                              const uint8_t *param_data) {
+  uint16_t pid = ntohs(header->param_id);
+  switch (ntohs(header->param_id)) {
+    case PID_DISPLAY_LEVEL:
+      return RDMResponder_GenericSetUInt8(header, param_data,
+                                          &g_moving_light.display_level);
+    default:
+      return RDM_RESPONDER_NO_RESPONSE;
+  }
+}
+
+int MovingLightModel_GetUint32(const RDMHeader *header,
+                               UNUSED const uint8_t *param_data) {
+  uint16_t pid = ntohs(header->param_id);
+  switch (ntohs(header->param_id)) {
+    case PID_DEVICE_HOURS:
+      return RDMResponder_GenericGetUInt32(header, g_moving_light.device_hours);
+    case PID_LAMP_HOURS:
+      return RDMResponder_GenericGetUInt32(header, g_moving_light.lamp_hours);
+    case PID_LAMP_STRIKES:
+      return RDMResponder_GenericGetUInt32(header, g_moving_light.lamp_strikes);
+    case PID_DEVICE_POWER_CYCLES:
+      return RDMResponder_GenericGetUInt32(
+          header, g_moving_light.device_power_cycles);
+    default:
+      return RDM_RESPONDER_NO_RESPONSE;
+  }
+}
+
+int MovingLightModel_SetUint32(const RDMHeader *header,
+                               const uint8_t *param_data) {
+  switch (ntohs(header->param_id)) {
+    case PID_DEVICE_HOURS:
+      return RDMResponder_GenericSetUInt32(header, param_data,
+                                           &g_moving_light.device_hours);
+    case PID_LAMP_HOURS:
+      return RDMResponder_GenericSetUInt32(header, param_data,
+                                           &g_moving_light.lamp_hours);
+    case PID_LAMP_STRIKES:
+      return RDMResponder_GenericSetUInt32(header, param_data,
+                                           &g_moving_light.lamp_strikes);
+    case PID_DEVICE_POWER_CYCLES:
+      return RDMResponder_GenericSetUInt32(header, param_data,
+                                           &g_moving_light.device_power_cycles);
+    default:
+      return RDM_RESPONDER_NO_RESPONSE;
+  }
+}
+
+int MovingLightModel_SetLampState(const RDMHeader *header,
+                                  const uint8_t *param_data) {
+  if (header->param_data_length != sizeof(uint8_t)) {
+    return RDMResponder_BuildNack(header, NR_FORMAT_ERROR);
+  }
+
+  if (param_data[0] > LAMP_STRIKE) {
```

Comment:
There are many values, including LAMP_ERROR, but we just allow setting to {0,1,2}